cloudinit/sources/DataSourceEc2.py

```diff
@@ -28,6 +28,8 @@
 STRICT_ID_PATH = ("datasource", "Ec2", "strict_id")
 STRICT_ID_DEFAULT = "warn"
 
+API_TOKEN_ROUTE = 'latest/api/token'
+
```

Comment:
```suggestion
AWS_TOKEN_TTL_SECONDS='21600'
```

Reply:
@blackboxsw and @raharper As I was testing this, I could see this token was being printed out in cloud-init.logs.
Also, Since the ttl is 6 hours, I could use the token and make calls to ec2 metadata service. Do you think its is right idea to : 

1) Make sure the token value is not printed in the logs?
2) If it has to be printed, Should this ttl value be low enough to make sure the token become unusable soon.

Reply:
> @blackboxsw and @raharper As I was testing this, I could see this token was being printed out in cloud-init.logs.
> Also, Since the ttl is 6 hours, I could use the token and make calls to ec2 metadata service. Do you think its is right idea to :
> 
> 1. Make sure the token value is not printed in the logs?
> 2. If it has to be printed, Should this ttl value be low enough to make sure the token become unusable soon.

Please file a bug and we can discuss this there.
[https://bugs.launchpad.net/cloud-init/+filebug](https://bugs.launchpad.net/cloud-init/+filebug)